apps/browser/components/Sidebar.tsx

```diff
@@ -62,19 +60,26 @@ const Sidebar = ({
         : "New session",
   }));
 
-  const createNewChat = async () => {
+  const { mutateAsync: createChat } = useCreateChat();
+  const { mutateAsync: deleteChat } = useDeleteChat();
+  const { mutateAsync: updateChat } = useUpdateChatTitle();
+  const [chatId] = useAtom(chatIdAtom);
+  const [isChatLoading, setIsChatLoading] = useAtom(isChatLoadingAtom);
+  const [workspace] = useAtom(workspaceAtom);
+
+  const workspaceUploadUpdate = useWorkspaceUploadUpdate();
+
+  const handleCreateNewChat = async () => {
     const id = uuid();
-    const createdChat = await createChat(id);
-    router.push(`/chat/${createdChat.id}`);
+    await createChat(id);
+    router.push(`/chat/${id}`);
+    setIsChatLoading(true);
```

Comment:
Is this actually needed? Can't we just initialize this as a `useState` on `app/page.tsx` and initialize it as true? The effect used there to make it false would turn it off on the first render; just like if the user accessed a specific chat directly via URL, right?

Reply:
Yes would be down to try this, did this to stop the app from flickering and entering into undesired states as the next page reroutes take significant time during development.

Reply:
Yeah I get you, because Next builds every page lazily at dev time; but if you `build` and `start` it might be instantaneous

Reply:
Yep exactly, lots of these bugs only appear in dev due to next.js loading and react strict mode